terraform/shared/shared.tf.json.template.py

```diff
@@ -59,6 +65,59 @@
                         }
                     ]
                 })
+            },
+            'config': {
+                # https://docs.aws.amazon.com/config/latest/developerguide/s3-bucket-policy.html
+                'bucket': '${aws_s3_bucket.config.id}',
+                'policy': json.dumps({
+                    'Version': '2012-10-17',
+                    'Statement': [
+                        {
+                            'Sid': 'AWSConfigBucketPermissionsCheck',
+                            'Effect': 'Allow',
+                            'Principal': {
+                                'Service': 'config.amazonaws.com'
+                            },
+                            'Action': 's3:GetBucketAcl',
+                            'Resource': f'arn:aws:s3:::{config.qualified_resource_name("config")}',
```

Comment:
Duplicative. You could have used a dedicated method in `Config` like we do for other bucket names or a TF reference. Put yourself in the shoes of someone wanting to rename the bucket: "config" is super generic and a textual search is gonna yield hundreds of hits. So would a search for `qualified_resource_name`. When you write code, please keep maintainability and your team mates in mind.

In this case you need to do both. 1) Add a config method and call it in the bucket resource definition. 2) make this a resource reference. Almost every TF resource has a .arn attribute that you can reference. On various occasions, we discussed why references are preferrable: they make the resource dependencies explicit. Without those, TF can't figure out the proper ordering of resource creation and destruction.

For resources other than buckets, the unqualified name should be `awsconfig`, not `config` since the latter is too generic.